app/Http/Controllers/LoginCredentialController.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\{
+    User,
+    LoginCredential
+};
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Str;
+use App\Http\Requests\LoginCredentialRequest;
+
+class LoginCredentialController extends Controller
+{
+    public function create(Request $request)
+    {
+        return view('login_credential.create');
+    }
+
+    public function store(LoginCredentialRequest $request)
+    {
+        $user = User::where('email', $request->email)->first();
+        if(!is_null($user) && !Hash::check($request->password,$user->password)){
+            return to_route('login_credential.create')->withErrors(['login_error' => 'パスワードが一致しません']);
+        }
+
+        if(!is_null($request->session('login_session_token'))){
+            $request->session()->forget('login_session_token');
+        }
+
+        $is_duplicated_login_session_token = true;
+        while($is_duplicated_login_session_token){
+            $login_session_token = Str::random(rand(30,50));
+            $is_duplicated_login_session_token = LoginCredential::where('token', $login_session_token)->exists();
+        }
+
+        LoginCredential::create([
+            'logged_in_at' => now(),
+            'user_id' => $user->id,
+            'token' => $login_session_token
+        ]);
+
+        $request->session()->put('login_session_token', $login_session_token);
+
+        return to_route('tasks.index');
+    }
+    
+    public function destroy(Request $request)
+    {
+        if(is_null($request->session('login_session_token'))){
+            return to_route('tasks.index');
+        }
+
+        $request->session()->forget('login_session_token');
+
+        return to_route('tasks.index');
```

Comment:
```suggestion
        if(!is_null($request->session('login_session_token'))){
            $request->session()->forget('login_session_token');
        }

        return to_route('tasks.index');
```
遷移先が同じであるなら、上記の方が短く書けます。

Reply:
ご指摘、ご提案ありがとうございます。
修正致しましたので、ご確認宜しくお願い致します。
```
        if(!is_null($request->session('login_session_token'))){
            $request->session()->forget('login_session_token');
        }

        return to_route('tasks.index');
```